### DISCUSSION TITLE
Driver: Failed to open the serial port: Error: No such file or directory, cannot open .....

### DISCUSSION THREAD

**Discussion by tarkani** [zwave-js/zwave-js-ui#3157] [Q&A] [answered]
**Config:**
Raspberry Pi4 Running on Debian GNU/Linux 11 (bullseye)
zwave-js-ui is running on **docker**
zwave-js-ui: 8.19.0.487524b
zwave-js: 11.0.0

**Z-Wave Controller:**
Aeotec Z-Stick 7 | Z-Wave Plus V2 | ZWA010

**Issue:**
Cannot open serial port. Exact message is:
` Driver: Failed to open the serial port: Error: No such file or directory, cannot open /dev/serial/by-id/usb-Silicon_Labs_CP2102N_USB_to_UART_Bridge_Controller_d8eb3e258e74ec11ac26a75019c2d21c-if00-port0 (ZW0100) `

**Goal:**
Enable Z-Wave integration with home assistant inside docker.

I installed docker and homeassistant on this device successfully. I also installed z-wave integration, following the instructions from this link: https://zwave-js.github.io/zwave-js-ui/#/getting-started/quick-start

One mistake I could have done is that I tried to do *all* 3 methods of installation as described in the "Quick Start" section. (Frankly, it doesn't say these are alternative methods. At least it is not clear for a novice like me).

The other interesting point is, my device name is extraordinarily long. (I also tried "/dev/ttyUSB0" with no luck.)

I also disabled "Soft Reset".

Any help is appreciated. My docker-compose.yml file looks like this:

`version: '3.7'

services:
  zwave-js-ui:
    container_name: zwave-js-ui
    image: zwavejs/zwave-js-ui:latest
    restart: always
    tty: true
    stop_signal: SIGINT
    environment:
      - SESSION_SECRET=mysupersecretkey
      - ZWAVEJS_EXTERNAL_CONFIG=/home/pi/docker/zwave-js-ui/store/.config-db
      # Uncomment if you want logs time and dates to match your timezone instead of UTC
      # Available at https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
      - TZ=Europe/Istanbul
    networks:
      - zwave
    devices:
      # Do not use /dev/ttyUSBX serial devices, as those mappings can change over time.
      # Instead, use the /dev/serial/by-id/X serial device for your Z-Wave stick.
      - '/dev/serial/by-id/usb-Silicon_Labs_CP2102N_USB_to_UART_Bridge_Controller_d8eb3e258e74ec11ac26a75019c2d21c-if00-port0:/dev/zwave'
    volumes:
      - zwave-config:/home/pi/docker/zwave-js-ui/store
    ports:
      - '8091:8091' # port for web interface
      - '3000:3000' # port for Z-Wave JS websocket server
networks:
  zwave:
volumes:
  zwave-config:
    name: zwave-config
`

**Comment by kpine** [comment, marked as answer]
1. Your USB path should be configured as `/dev/zwave` in the ZUI config, as that is the path you have configured with Docker.
2. You shouldn't disable Soft Reset for a 700-series controller in Docker.
3. Your store volume is wrong. Use the examples. Don't set, or don't change, ZWAVEJS_EXTERNAL_CONFIG either.

**Reply by tarkani** [reply]
Thanks! This resolved my issue. 

I had changed directories for store and ZWAVEJS_EXTERNAL_CONFIG because /usr/src/app doesn't exist. I thought it would fail because of that. Apparently I was wrong.

Thanks again!

**Reply by kpine** [reply]
To clarify, `/usr/src/app/store` is the path inside the container, not on the host. Z-Wave JS uses that directory unless overridden (which it was not).

This:
```yaml
volumes:
- zwave-config:/usr/src/app/store
```
uses a [Docker volume](https://docs.docker.com/storage/volumes/) named `zwave-config` and maps the volume directory to the container path `/usr/src/app/store`. The host volume directory is somewhere in your Docker data directory (usually `/var/lib/docker`).

This:
```yaml
volumes:
- /home/pi/docker/zwave-js-ui/store:/usr/src/app/store
```
creates a [Docker bind mount](https://docs.docker.com/storage/bind-mounts/) mapping the host directory `/home/pi/docker/zwave-js-ui/store` to the container path `/usr/src/app`. This is probably what you originally wanted.
